### PR TITLE
Step 1: Execute python script from Java

### PR DESCRIPTION
**Pull request by sophiesongge**

### CODE REVIEW

File: src/main/java/EntryPoint/EntryPoint.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package EntryPoint;
```
Comment (fviale) [inline src/main/java/EntryPoint/EntryPoint.java]: better to define a common package for your script engine, for example: jsr223.cpython
Also, package names are lowercase in Java

File: src/main/java/python/PythonScriptWriter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        File pythonTempFile = null;
+        try {
+            pythonTempFile = File.createTempFile("jsr223-cpython-", PYTHON_FILE_EXTENSION);
+        } catch (IOException e) {
+            throw new IOException("Unable to create python temp file. " + e);
+        }
+
+        // Write python script file to disk
+        Writer pythonScriptFileWriter = new FileWriter(pythonTempFile);
+        pythonScriptFileWriter.write("from py4j.java_gateway import JavaGateway, GatewayParameters" + "\n");
```
Comment (fviale) [inline src/main/java/python/PythonScriptWriter.java]: Better to use a portable stream allowing to call writeLine

**Comment by fviale** [comment]
I think in general it is cleaner to have two script engines for cpython 2 and cpython 3. As the language syntax is different, the workflow designer must be able to specify it. This decision, should not be for example defined in a java property on the node.
Of course it possible to factorize most of the code, and still declare two script engines: cpython, cpython3

File: src/main/java/ProcessBuilder/SingletonPythonProcessBuilderFactory.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * or a different license than the AGPL.
+ */
+package ProcessBuilder;
+
+/**
+ * @author ActiveEon Team
+ * @since 05/10/2017
+ */
+public class SingletonPythonProcessBuilderFactory implements PythonProcessBuilderFactory {
+    private static SingletonPythonProcessBuilderFactory ourInstance = new SingletonPythonProcessBuilderFactory();
```
Comment (ShatalovYaroslav) [inline src/main/java/ProcessBuilder/SingletonPythonProcessBuilderFactory.java]: I suppose the factory should also 'final' to be thread safe automatically.
Please check  https://en.wikipedia.org/wiki/Singleton_pattern

File: src/main/java/python/PythonCommandCreator.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 05/10/2017
+ */
+public class PythonCommandCreator {
+
```
Comment (ShatalovYaroslav) [inline src/main/java/python/PythonCommandCreator.java]: I think the versions should not be hardcoded

**Review by ShatalovYaroslav — APPROVED** [review]
In general LGTM, except the changes specified in Fabien's comment

File: src/main/java/ProcessBuilder/SingletonPythonProcessBuilderFactory.java (hunk 3 above)
Comment (sophiesongge) [reply src/main/java/ProcessBuilder/SingletonPythonProcessBuilderFactory.java]: Yes you're right, it's automatically generated by Intellij and I didn't check it. Thanks for the comments and it's changed

File: src/main/java/python/PythonCommandCreator.java (hunk 4 above)
Comment (sophiesongge) [reply src/main/java/python/PythonCommandCreator.java]: This will be changed in step 2

File: src/main/java/EntryPoint/EntryPoint.java (hunk 1 above)
Comment (sophiesongge) [reply src/main/java/EntryPoint/EntryPoint.java]: done

File: src/main/java/python/PythonScriptWriter.java (hunk 2 above)
Comment (sophiesongge) [reply src/main/java/python/PythonScriptWriter.java]: done

File: src/main/java/jsr223/cpython/python/PythonScriptWriter.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        try {
+            pythonTempFile = File.createTempFile("jsr223-cpython-", PYTHON_FILE_EXTENSION);
+        } catch (IOException e) {
+            throw new IOException("Unable to create python temp file. " + e);
+        }
+
+        // Write python script file to disk
+        FileWriter pythonScriptFileWriter = new FileWriter(pythonTempFile);
+        BufferedWriter pythonScriptBufferedWriter = new BufferedWriter(pythonScriptFileWriter);
+        try {
```
Comment (fviale) [inline src/main/java/jsr223/cpython/python/PythonScriptWriter.java]: I think you can use the try with ressources syntax here

**Pull request merged**